src/native/external/zlib-intel/deflate.c

```diff
@@ -1484,10 +1484,10 @@ local block_state deflate_stored(s, flush)
         _tr_stored_block(s, (char *)0, 0L, last);
 
         /* Replace the lengths in the dummy stored block with len. */
-        s->pending_buf[s->pending - 4] = len;
-        s->pending_buf[s->pending - 3] = len >> 8;
-        s->pending_buf[s->pending - 2] = ~len;
-        s->pending_buf[s->pending - 1] = ~len >> 8;
+        s->pending_buf[s->pending - 4] = (Bytef)len;
```

Comment:
I wonder if we should touch sources of the external libraries unless we are planning to backport the change upstream. This could cause some pain if we update it in the future.

Reply:
We've been doing this for libunwind too. I submitted an upstream PR: https://github.com/libunwind/libunwind/pull/333. I'm fine doing either, just let me know.

Reply:
Based on https://github.com/dotnet/runtime/pull/68615#issuecomment-1111412946, I'm just going to revert these changes and suppress the warning in that project.